OpenRA.Mods.D2k/Traits/Buildings/DamagedWithoutFoundation.cs

```diff
@@ -52,7 +52,10 @@ public void AddedToWorld(Actor self)
 					safeTiles++;
 			}
 
-			damageThreshold = (info.DamageThreshold * health.MaxHP + (100 - info.DamageThreshold) * safeTiles * health.MaxHP / totalTiles) / 100;
+			if (totalTiles > 0)
+				damageThreshold = (info.DamageThreshold * health.MaxHP + (100 - info.DamageThreshold) * safeTiles * health.MaxHP / totalTiles) / 100;
+			else
+				damageThreshold = health.HP;
```

Comment:
No they wont.
```
			var delta = health.HP - damageThreshold;
			if (delta > 0)
				health.InflictDamage(self, self.World.WorldActor, delta, null, false);
```
the delta is the damage that will be taken so if damageThreshold is equal to health.HP then delta will be 0 and no damage will happen.